### PR TITLE
Fix some deprecations and other warnings

### PR DESCRIPTION
**Pull request by marco-neumann-jdas**
See commit msgs for details.

There is one notable case left where we test the roundtrip of byte-string columns through pyarrow, likely a leftover from python 2. Since I would like to raise in the serializer in that case and also adjust the tests, this will be done in another PR.

### CODE REVIEW

File: kartothek/serialization/_generic.py [hunk 1]
```diff
@@ -378,7 +378,7 @@ def _ensure_type_stability(
         error messages.
     """
 
-    value_dtype = pd.Series(value).dtype
+    value_dtype = pd.Series(value if is_list_like(value) else [value]).dtype
```
Comment (fjetter) [inline kartothek/serialization/_generic.py:381]: Is this user facing, i.e. is it relevant enough to mention in a changelog?

Comment (marco-neumann-jdas) [reply kartothek/serialization/_generic.py:381]: Do we mention fixed warnings in the changelog? I'm OK with this, just wanted to clarify.

Comment (fjetter) [reply kartothek/serialization/_generic.py:381]: If it's just about the warning, I don't care at all. I thought it was a fix since you added another test

Comment (marco-neumann-jdas) [reply kartothek/serialization/_generic.py:381]: Valid assumption/question. Added the other test to make sure it is working (like a safety net). The test status didn't change due to the actual change here.

**Pull request merged**